infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java

```diff
@@ -115,34 +141,64 @@ public void sendResponse(ChannelHandlerContext ctx, CorfuMsg inMsg, CorfuMsg out
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) {
         try {
-            // The incoming message should have been transformed to a CorfuMsg earlier in the
-            // pipeline.
-            CorfuMsg m = ((CorfuMsg) msg);
-            // We get the handler for this message from the map
-            AbstractServer handler = handlerMap.get(m.getMsgType());
-            if (handler == null) {
-                // The message was unregistered, we are dropping it.
-                log.warn("Received unregistered message {}, dropping", m);
-            } else {
-                if (messageIsValid(m, ctx)) {
-                    // Route the message to the handler.
-                    if (log.isTraceEnabled()) {
-                        log.trace("Message routed to {}: {}", handler.getClass().getSimpleName(), msg);
+            if (msg instanceof CorfuMsg) {
+                // The incoming message should have been transformed to a CorfuMsg earlier in the pipeline.
+                CorfuMsg m = ((CorfuMsg) msg);
+                // We get the handler for this message from the map
+                AbstractServer handler = handlerMap.get(m.getMsgType());
+                if (handler == null) {
+                    // The message was unregistered, we are dropping it.
+                    log.warn("channelRead: Received unregistered message {}, dropping", m);
+                } else {
+                    if (messageIsValid(m, ctx)) {
```

Comment:
Since you have deprecated a bunch  of methods in `IServerRouter`, I'd say it's better not to add new methods into IServerRouter but create another interface `ProtobufServerRouter` wich will contain all those new methods and in your code... there will be an "if"  condition that defines which router to use

Reply:
We had an offline discussion, an got to agrrement